### ISSUE TITLE
Potential memory leak with `BatchNorm`

### ISSUE THREAD

**Issue by genesis-jamin** [e3nn/e3nn#414]
Hi there, I wanted to bring to your attention a potential memory leak in the BatchNorm implementation. I am not familiar with the mathematics behind equivariant neural networks and so I can't say if the theory demands this implementation, but it seems like the running mean / variance will grow endlessly large as we keep training because a new tensor is concatenated every batch. Is my understanding here correct or am I missing something? Thanks!

https://github.com/e3nn/e3nn/blob/bb71b3c75c38c6d633cfebe9988b1fb1563432c0/e3nn/nn/_batchnorm.py#L190-L194

**Comment by mariogeiger** [comment]
`torch.cat` does not append to `out`

```
x = torch.randn(20, 3)

torch.cat([torch.randn(10, 3), torch.randn(10, 3)], out=x)
print(x.shape)  # torch.Size([20, 3])

torch.cat([torch.randn(10, 3), torch.randn(10, 3)], out=x)
print(x.shape)  # torch.Size([20, 3])

```
